### ISSUE TITLE
http localhost:3000 has been blocked CORS

### ISSUE THREAD

**Issue by henjoe** [VolkovLabs/volkovlabs-form-panel#55]
Hello, I actually get stocked by this error.
![image](https://user-images.githubusercontent.com/49782655/177084809-0f2f1ce4-1df2-411d-8ecb-5e1b88b093a5.png)

I am a NodeRed running on the same machine with Grafana. I am using nodered as may API, I simply want to get what the user inputted in my grafana to my nodered API. 
However, I always get error as mentioned above.
In addition, I have a basic authentication on my nodered API, but I don't know where should I add those authentication from Data Manipulation Panel.

Is there a configuration or settings that I need to properly config it?

Thank you.

**Comment by mikhail-vl** [comment]
Hello @henjoe, thank you for submitting the issue.

  Could you please take a look at our article "How to connect the Data Manipulation plugin for Grafana to API Server" (https://volkovlabs.com/how-to-connect-the-data-manipulation-plugin-for-grafana-to-api-server-1abe5f60c904), which explains the CORS error you are getting.
  The plugin repository has an example of the Node.js server which allows CORS and should be similar to NodeRed: https://github.com/VolkovLabs/volkovlabs-form-panel/blob/main/server/server.ts

  To configure Basic Authentication you will need to add Header Parameter:
  - Name: `Authorization`
  - Value: `Basic amXXXXX=` with base64-encoded credentials.

  Please let me know if there is anything else.

**Comment by henjoe** [comment]
Hi @mikhail-vl  Thanks for your response. 
I will try those link you have sent, and will get back to you once done. 

Again, thank you for your reply. 

Regards,
Henjoe

**Comment by mikhail-vl** [comment]
@henjoe Please keep us updated with your progress!

**Comment by henjoe** [comment]
Hi @mikhail-vl ,

Upon testing the link you have sent. I used the 2nd options which is using the Nginx reverse proxy which I think it work. 
I can access my grafana on port 443 when I access my localhost. 
No more related Cors error, however, here's what I am encountering right now.

![image](https://user-images.githubusercontent.com/49782655/177237602-62222ca0-37fc-43e9-9f49-c2ff2d4b97c2.png)

As above screen shot, it's kinda related to "web socket connection". 

Below is my configuration for my nginx, I just combined configuration from your Blog as well as the documentation from grafana.
![image](https://user-images.githubusercontent.com/49782655/177237568-f5931937-dd22-4d99-9cba-aff2e0f80a4e.png)

Do I have misconfiguration on the side of Nginx?
Hoping for your guidance. 

Thank you.

Regards,
Henjoe
